### PR TITLE
Allow in memory manipulation of nalu yaml files

### PR DESCRIPTION
**Pull request by psakievich**
Allow users to overwrite entries in a single template nalu-wind yaml file for each yaml instance.  
This PR allows you to specify parameters to change in all nalu-wind instance and each one individually. See the added test file for examples. 

This PR only allows you to replace existing leaf nodes in the yaml graph so you can over write values, but you can not add totally new nodes to the graph i.e. add in a mesh motion if it doesn't exist in the template file. 

It will change the naming convention for the log files to be `[input_file]_[index].log` when this is employed rather than `[input_file].log`.  

I have unit-tests for this in a stand alone project.  When we hook up googletest I can copy them over to the driver repo.

### CODE REVIEW

**Comment by ashesh2512** [comment]
@psakievich Do we have the option to write the various nalu files for debugging purposes? Did I miss that somewhere in the PR?

**Comment by psakievich** [comment]
Yes it is in the regression test.

**Comment by psakievich** [comment]
@ashesh2512 I currently have it as `write_final_yaml_to_disk: true` and then it will take the logfile name for that instance and swap the suffix to be `.yaml` instead of `.log`.  That way they are paired together.

But the more I think about it the more I want to make nalu just natively dump the input file into the logfile. Then you can see exactly what options were used for the run AND run the reproducer all by just having someone send you the log file.

File: app/exawind/exawind.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #include "OversetSimulation.h"
 #include "MPIUtilities.h"
 #include "mpi.h"
+#include "yaml-editor.hpp"
```
Comment (marchdf) [inline app/exawind/exawind.cpp:6]: this is minor but... we don't really use the `hpp` extension in the exawind projects. Just noting in case someone cares

**Review by marchdf — APPROVED** [review]

File: app/exawind/exawind.cpp (hunk 1 above)
Comment (psakievich) [reply app/exawind/exawind.cpp:6]: Good call. I didn't think about that.

**Pull request merged**